[Word] (annotations) Update article (#5149)
OfficeDev/office-js-docs-pr: docs/word/annotations.md

---
title: Use annotations in your Word add-in
description: Learn to use annotations in your Word add-in.
ms.date: 04/29/2025
ms.localizationpriority: medium
ms.topic: how-to
---

# Use annotations in your Word add-in

> **Includes community contributions from:** Abdulhadi Jarad

Reviewer: ```suggestion
**Includes community contributions from:** [Abdulhadi Jarad](https://github.com/abdulhadi-jarad)
```

> You can use annotations to provide feedback about grammar or other aspects of content in a Word document. The user may see colorful underlining that indicates there's an issue or other information. If the user hovers over the affected content, a popup dialog is displayed that should show the user what the issue is and possible actions they can take.

Reviewer: ```suggestion
You can use annotations to provide feedback about grammar or other aspects of content in a Word document. The user may see colorful underlining that indicates there's an issue or other information. If the user hovers over the affected content, a popup dialog is displayed that shows them what the issue is and possible actions they can take.
```

  - [CritiqueAnnotation](/javascript/api/word/word.critiqueannotation): Represents the critique type of annotation.
  - [Critique](/javascript/api/word/word.critique): Represents feedback about an affected area of a paragraph, indicated by a colored underline.
- Annotation events on [Document](/javascript/api/word/word.document):
  - [onAnnotationClicked](/javascript/api/word/word.document#word-word-document-onannotationclicked-member)
  - [onAnnotationHovered](/javascript/api/word/word.document#word-word-document-onannotationhovered-member)
  - [onAnnotationInserted](/javascript/api/word/word.document#word-word-document-onannotationinserted-member)
  - [onAnnotationPopupAction](/javascript/api/word/word.document#word-word-document-onannotationpopupaction-member)
  - [onAnnotationRemoved](/javascript/api/word/word.document#word-word-document-onannotationremoved-member)

## Use annotation APIs

The following sections show how to work with annotation APIs. These examples are based on the [Manage annotations](https://github.com/OfficeDev/office-js-snippets/blob/prod/samples/word/50-document/manage-annotations.yaml) code sample.

> In your add-in, your code should manage annotations more dynamically according to feedback or critique results from your service run against the user's content in the document.

Reviewer: ```suggestion
Your add-in code should use the feedback or critique results from your service run against the user's content in the document to manage annotations more dynamically.
```

```typescript
let eventContexts = [];

async function registerEventHandlers() {
  // Registers event handlers.
  await Word.run(async (context) => {
    eventContexts[0] = context.document.onParagraphAdded.add(paragraphChanged);
    eventContexts[1] = context.document.onParagraphChanged.add(paragraphChanged);

    eventContexts[2] = context.document.onAnnotationClicked.add(onClickedHandler);
    eventContexts[3] = context.document.onAnnotationHovered.add(onHoveredHandler);
    eventContexts[4] = context.document.onAnnotationInserted.add(onInsertedHandler);
    eventContexts[5] = context.document.onAnnotationRemoved.add(onRemovedHandler);
    eventContexts[6] = context.document.onAnnotationPopupAction.add(onPopupActionHandler);

    await context.sync();

    console.log("Event handlers registered.");
  });
}
```

### onClickedHandler event handler

The following code runs when the registered `onAnnotationClicked` event occurs.

```typescript
async function onClickedHandler(args: Word.AnnotationClickedEventArgs) {

>   await Word.run(async (context) => {

Reviewer: For consistency, do you want to add a remark here and in the next four samples?

Author: Added

    for (let i = annotations.items.length - 1; i >= 0; i--) {
      const annotation: Word.Annotation = annotations.items[i];

      if (annotation.state === Word.AnnotationState.created) {
        console.log(`Rejecting ID ${annotation.id}...`);
        annotation.critiqueAnnotation.reject();

        await context.sync();
        break;
      }
    }
  });
}
```

## Delete annotations

The following code shows how to delete all the annotations found in the selected paragraph.

```typescript
async function deleteAnnotations() {
  // Deletes all annotations found in the selected paragraph.
  await Word.run(async (context) => {
    const paragraph: Word.Paragraph = context.document.getSelection().paragraphs.getFirst();
    const annotations: Word.AnnotationCollection = paragraph.getAnnotations();
    annotations.load("id");

    await context.sync();

    const ids = [];
    for (let i = 0; i < annotations.items.length; i++) {
      const annotation: Word.Annotation = annotations.items[i];

      ids.push(annotation.id);
      annotation.delete();
    }

    await context.sync();

    console.log("Annotations deleted:", ids);
  });
}
```

## Deregister annotation events

> the following code shows how to deregister the event handlers using their event contexts you tracked in the `eventContext` variable.

Reviewer: ```suggestion
The following code shows how to deregister the event handlers using their event contexts you tracked in the `eventContext` variable.
```
